internal/transform/ledger.go

```diff
@@ -113,18 +114,39 @@ func extractCounts(lcm xdr.LedgerCloseMeta) (transactionCount int32, operationCo
 	transactionCount = int32(txCount) - failedTxCount
 	txSetOperationCount = strconv.FormatInt(int64(txSetOperationCounter), 10)
 	return
+
 }
 
-func getTransactionProcessing(ledgerCloseMeta xdr.LedgerCloseMeta) (transactionProcessing []xdr.TransactionResultMeta) {
-	switch ledgerCloseMeta.V {
+func getTransactionSet(transactionEntry historyarchive.Ledger) (transactionProcessing []xdr.TransactionEnvelope) {
+	switch transactionEntry.Transaction.Ext.V {
 	case 0:
-		return ledgerCloseMeta.V0.TxProcessing
+		return transactionEntry.Transaction.TxSet.Txs
 	case 1:
-		return ledgerCloseMeta.V1.TxProcessing
-	case 2:
-		return ledgerCloseMeta.V2.TxProcessing
+		return getTransactionPhase(transactionEntry.Transaction.Ext.GeneralizedTxSet.V1TxSet.Phases)
 	default:
-		panic(fmt.Sprintf("Unsupported LedgerCloseMeta.V: %d", ledgerCloseMeta.V))
+		panic(fmt.Sprintf("Unsupported TransactionHistoryEntry.Ext: %d", transactionEntry.Transaction.Ext.V))
 	}
 }
 
+func getTransactionPhase(transactionPhase []xdr.TransactionPhase) (transactionEnvelope []xdr.TransactionEnvelope) {
+	transactionSlice := []xdr.TransactionEnvelope{}
+	for _, phase := range transactionPhase {
+		switch phase.V {
+		case 0:
+			for _, component := range *phase.V0Components {
```

Comment:
nit: Could use the xdr Must/GetV0Components() functions as an extra form of checking if xdr elements are correct or not